src/sentry/incidents/endpoints/organization_alert_rule_details.py

```diff
@@ -58,10 +58,10 @@ def fetch_alert_rule(request: Request, organization, alert_rule):
                     action.get("_sentry_app_installation", {}).get("sentry_app_id", None)
                 )
     if sentry_app_ids:
-        sentry_app_map = {
-            install.sentry_app.id: install.sentry_app
-            for install in app_service.get_many(filter=dict(app_ids=sentry_app_ids))
-        }
+        fetched_apps = app_service.get_many(
+            filter=dict(app_ids=sentry_app_ids, organization_id=organization.id)
```

Comment:
This seems reasonable. Do we need any additional tests to verify that this behavior is correct?

Reply:
Yup working on that now